### PR TITLE
LPS-94575 DateInput and DateTimeInput on IE11

### PR DESCRIPTION
**Pull request by andresfulla**

### CODE REVIEW

**Comment by andresfulla** [comment]
Hi @bryceosterhaus , this is one is rebased on top of our ongoing pull https://github.com/bryceosterhaus/liferay-portal/pull/43.

To fix misbehaviours on IE11 with the `<input type="date" />` we're using an internal state, and we only propagate the change event when the user blurs the input. To keep the input always up to date we're updating the internal state whenever the prop changes.

Thanks

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateInput.es.js [hunk 1]
```diff
@@ -9,31 +9,73 @@ class DateInput extends React.Component {
...
+		super(props);
+
+		const date = new Date(props.value);
+
+		this.state = {
+			value: jsDatetoYYYYMMDD(date)
+		};
+	}
+
+	shouldComponentUpdate(nextProps) {
```
Comment (bryceosterhaus) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateInput.es.js]: You should probably use `getDerivedStateFromProps` for this lifecycle. https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateTimeInput.es.js [hunk 2]
```diff
@@ -11,35 +11,76 @@ class DateTimeInput extends React.Component {
...
+		super(props);
+
+		const date = new Date(props.value);
+
+		this.state = {
+			value: dateFns.format(date, INPUT_DATE_FORMAT)
+		};
+	}
+
+	shouldComponentUpdate(nextProps) {
```
Comment (bryceosterhaus) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateTimeInput.es.js]: You can use https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops to get around having to do this logic in both the constructor and shouldComponentUpdate

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateInput.es.js (hunk 1 above)
Comment (andresfulla) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateInput.es.js]: Thanks!! This is exactly what we need.

**Comment by andresfulla** [comment]
Hi @bryceosterhaus, thanks for the tip on derived state from props. It's using it now. I introduced the `initialValue` on the state, so it's able to compare with the previous version of `props.value`.

**Comment by bryceosterhaus** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/72977. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**